packages/superset-ui-core/src/utils/removeDuplicates.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/**
+ * Remove duplicate items from a list.
+ */
+export function removeDuplicates<T>(items: T[], hash?: (item: T) => unknown): T[] {
+  const seen = new Set();
+  const result: T[] = [];
+  items.forEach(x => {
+    const itemHash = hash ? hash(x) : x;
+    if (!seen.has(itemHash)) {
+      seen.add(itemHash);
+      result.push(x);
+    }
+  });
+  return result;
+}
```

Comment:
may be only `Set` enougph? if I got correctly logic do it like (but not sure):
```
items.forEach(x => {
    if (!hash || !hash(x));
      seen.add(x);
    }
  });
return [...seen]
```

Reply:
Does set spread preserve orders?

Reply:
[MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#iteration_methods) says yes, but this still wouldn't work for my needs because I need to check uniqueness by hash, not the item itself.

Reply:
ideas: can short circuit when `hash` is not used and only check `hash` once.

```ts
function removeDuplicates<T>(items: T[], hash?: (item: T) => unknown): T[] {
  if (hash) {
    const seen = new Set();
    return items.filter(x => {
      const itemHash = hash(x);
      if (seen.has(itemHash)) return false; 
      seen.add(itemHash);
      return true;
    });
  }

  return [...new Set(items)];
}
```

Reply:
Good idea! Updated.